### ISSUE TITLE
Error Compiling on macOS Sierra: invalid operands to binary expression

### ISSUE THREAD

**Issue by datatypevoid** [Arthmoor/AFKMud#18]
Compiling on macOS Sierra version 10.12.2 gives me the output:

```
Building AFKMud....
/Applications/Xcode.app/Contents/Developer/usr/bin/make -j2 -s afkmud
  Compiling o/comm.o....
  Compiling o/comments.o....
clang: clang: warning: argument unused during compilation: '-rdynamic'
clang: warning: argument unused during compilation: '-e xport-dynamic'
warning: argument unused during compilation: '-rdynamic'
clang: warning: argument unused during compilation: '-e xport-dynamic'
warning: warning: unknown warning option '-Wuseless-cast' [-Wunknown-warning-option]unknown
 warning option '-Wuseless-cast' [-Wunknown-warning-option]
warning: unknown warning option '-Wstrict-null-sentinel' warning[-Wunknown-warning-option]:
unknown warning option '-Wstrict-null-sentinel' [-Wunknown-warning-option]
comm.cpp:377:59: error: invalid operands to binary expression ('__bind<int &, sockaddr *, unsigned long>' and 'int')
   if( bind( fd, ( struct sockaddr * )&sa, sizeof( sa ) ) == -1 )
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~
2 warnings and 1 error generated.
make[1]: *** [o/comm.o] Error 1
make[1]: *** Waiting for unfinished jobs....
2 warnings generated.
make: *** [all] Error 2
```

**Comment by datatypevoid** [comment]
Tried to register on the SmaugMUD site to post on the support section of the forums as well, and I seem to be stuck with an unactivated account, even though I went through the activation process several times.

**Comment by brazilianldsjaguar** [comment]
Was running into the same problem. It boiled down to three things:

1. Adding a `#include <sys/socket.h>` up top (you could wrap this in a `#if defined(__APPLE__)` directive if you want)
2. Changing the function from:
    `bind( fd, ( struct sockaddr * )&sa, sizeof( sa ) )`
to
    `bind( fd, ( const struct sockaddr *_&sa, (socklen_t)sizeof( sa ) )`

The main difference being that you're using a 'const struct' and casting to `socklen_t`, which allows the compiler to properly match the `bind()` signature found in `socket.h`

**Comment by brazilianldsjaguar** [comment]
I know this is closed, but I just wanted to mention I made a fix for Mac OSX (Sierra 10.12.6 specifically) here (https://github.com/Arthmoor/AFKMud/pull/20).
